lib/common/api/deprecate.js

```diff
@@ -69,23 +69,28 @@ const deprecate = {
     })
   },
 
-  promisify: (promise, cb) => {
-    const oldName = `function with callbacks`
-    const newName = `function with Promises`
-    const warn = warnOnce(oldName, newName)
+  promisify: (fn, numArgs) => {
+    return function (...args) {
+      const promise = fn.apply(this, args.splice(0, numArgs))
+      const cb = args[numArgs]
```

Comment:
_(bug)_ wouldn't this be `args[0]` after the previous line's splice?

Reply:
_(tweak)_ Also, since we've coded this to the assumption that `cb` always the final parameter... why not just `.pop()` it off of args and drop the `numArgs` parameter, e.g.

```
promisify: fn => {
  return function (...args) {
    const cb = args.pop();
    const promise = fn.apply(this, args);
```

Reply:
it's not fully equivalent to the original callback based function, for example if you have `contents.printToPDF(options, callback)` and call it with `contents.printToPDF({}, () => console.log('done'), () =>  console.log('should not be called'));`, then the extra arguments are not ignored as they were before.

Reply:
OK, that's fair.

Is that something that happens in practice? Should we warn when people pass too many arguments?

Reply:
@ckerr Passing too many arguments is not considered an error / warning.  It can happen if you're doing things like blindly spreading args `...args`.  We do already error out in native land if you provide invalid arguments.